### PR TITLE
Do not report a secondary NFW exception after already reporting one earlier.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Workspaces/Core/Portable/Workspace/Host/SyntaxTreeFactory/AbstractSyntaxTreeFactoryService.cs [hunk 1]
```diff
@@ -9,6 +9,7 @@
 using System.IO;
 using System.Text;
 using System.Threading;
+using Microsoft.CodeAnalysis.Options;
```
Comment (genlu) [inline src/Workspaces/Core/Portable/Workspace/Host/SyntaxTreeFactory/AbstractSyntaxTreeFactoryService.cs:12]: using seem unnecessary

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Workspace/Host/SyntaxTreeFactory/AbstractSyntaxTreeFactoryService.cs:12]: indeed.

Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Workspace/Host/SyntaxTreeFactory/AbstractSyntaxTreeFactoryService.cs:12]:
```suggestion
```

File: src/Workspaces/Remote/Core/RemoteCallback.cs [hunk 2]
```diff
@@ -129,34 +129,21 @@ private static bool ReportUnexpectedException(Exception exception, CancellationT
...
-            cancellationToken.ThrowIfCancellationRequested();
-
-            if (exception is ConnectionLostException)
-            {
-                throw new OperationCanceledIgnoringCallerTokenException(exception);
-            }
-
-            // If this is hit the cancellation token passed to the service implementation did not use the correct token,
-            // and the resulting exception was not a ConnectionLostException.
-            return ExceptionUtilities.Unreachable();
```
Comment (genlu) [inline src/Workspaces/Remote/Core/RemoteCallback.cs:159]: Is this the cause of the second NFW?

**Review by genlu — APPROVED** [review]

File: src/Workspaces/Remote/Core/RemoteCallback.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Remote/Core/RemoteCallback.cs:159]: Yup yup!

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski  going to still wait on you for this to make sure i'm ok on this.

**Review by arkalyanms — APPROVED** [review]

**Review by jasonmalinowski — APPROVED** [review]

**Pull request merged**